platforms/ios/lib/WysiwygComposer/Sources/WysiwygComposer/Components/WysiwygComposerView/WysiwygComposerViewModel.swift

```diff
@@ -378,13 +419,50 @@ public extension WysiwygComposerViewModel {
             }
             plainTextContent = textView.attributedText
         } else {
-            reconciliateIfNeeded()
+            let dotInserted = checkForDoubleSpaceToDotConversion()
+            if !dotInserted {
+                reconciliateIfNeeded()
+            }
             applyPendingFormatsIfNeeded()
         }
         
         updateCompressedHeightIfNeeded()
     }
     
+    func checkForDoubleSpaceToDotConversion() -> Bool {
+        let text = textView.attributedText.htmlChars.withNBSP
+        guard text.count > 0 else {
+            return false
+        }
+        let content = attributedContent.text.htmlChars.withNBSP
+        let dotStart = textView.selectedRange.location - 1
+        let dotEnd = textView.selectedRange.location
+        guard dotStart >= 0, dotEnd <= text.count else {
+            return false
+        }
+        let dotStartIndex = text.index(text.startIndex, offsetBy: dotStart)
+        let dotEndIndex = text.index(after: dotStartIndex)
+        guard
+            dotStartIndex < text.endIndex,
+            dotEndIndex <= text.endIndex,
+            dotStartIndex < content.endIndex,
+            dotEndIndex <= content.endIndex
+        else {
+            return false
+        }
+        let dotRange = dotStartIndex..<dotEndIndex
+        let textPotentialDot = String(text[dotRange])
+        let contentPotentialDot = String(content[dotRange])
+        if textPotentialDot == ".", contentPotentialDot != "." {
+            let replaceUpdate = model.replaceTextIn(newText: ".",
+                                                    start: UInt32(dotStart),
+                                                    end: UInt32(dotEnd))
+            applyUpdate(replaceUpdate, skipTextViewUpdate: true)
+            return true
```

Comment:
I have now seen that tis function actually also modifies the content... okay... at this point instead of having it has a property... I would keep it as a function.... and have it return the `replaceUpdate` as an optional.
Would avid having a function like this have a collateral effect.

so it would be something like 
```
if let update = doubleSpaceToDotConversionUpdate() {
	applyUpdate(replaceUpdate, skipTextViewUpdate: true)
	reconciliateIfNeeded()
} 
```

when used inside the replace text function

Reply:
I'm not sure I follow where the above code is supposed to go?  The func `replaceText(range: NSRange, replacementText: String)` fuction? 

I believe`reconciliateIfNeeded` and ` doubleSpaceToDotConversion` need to be called from `didUpdateText`. `didUpdateText` is called after the text view is updated. After it's updated we are checking if the textView was changed from under us (via 2 spaces => dot or some other change to the text view that requires the string differ/reconciliate)

Reply:
yeah they should go in the `didUpdateText` (my bad mistyped the name) the idea is just to turn the function into a simple function that returns an optional, and the apply the  update in the `didUpdateText`.
This is because this function returns a boolean but is also applying changes in the textView, which is not clear from the name and intent of the function

Reply:
I've just handled the new case of the dot after the inline text prediction, which is similar but different to the two space to dot conversion.

I've kept the `applyUpdate` inside the functions as it is intrinsically linked to the model update so I think should be kept together to handle each special case. I've updated function names and also added docs, so hopefully its clear to the calling site.